src/libData/AccountData/AccountStore.cpp

```diff
@@ -175,45 +175,74 @@ bool AccountStore::MoveUpdatesToDisk() {
   unique_lock<shared_timed_mutex> g(m_mutexPrimary, adopt_lock);
   lock_guard<mutex> g2(m_mutexDB, adopt_lock);
 
-  unordered_map<string, string> batch;
+  unordered_map<string, string> code_batch;
+  unordered_map<string, string> initData_batch;
 
   for (auto i : *m_addressToAccount) {
     if (i.second.isContract()) {
       if (ContractStorage::GetContractStorage()
               .GetContractCode(i.first)
               .empty()) {
-        batch.insert({i.first.hex(),
-                      DataConversion::CharArrayToString(i.second.GetCode())});
+        code_batch.insert({i.first.hex(), DataConversion::CharArrayToString(
+                                              i.second.GetCode())});
+        initData_batch.insert({i.first.hex(), DataConversion::CharArrayToString(
+                                                  i.second.GetInitData())});
       }
     }
   }
 
-  if (!ContractStorage::GetContractStorage().PutContractCodeBatch(batch)) {
+  if (!ContractStorage::GetContractStorage().PutContractCodeBatch(code_batch)) {
     LOG_GENERAL(WARNING, "PutContractCodeBatch failed");
     return false;
   }
 
-  if (!HASHMAP_CONTRACT_STATE_DB) {
-    ContractStorage::GetContractStorage().GetStateDB().commit();
+  bool ret = true;
 
-    for (auto i : *m_addressToAccount) {
-      i.second.Commit();
+  if (!ContractStorage::GetContractStorage().PutContractInitDataBatch(
+          initData_batch)) {
+    LOG_GENERAL(WARNING, "PutContractInitDataBatch failed");
+    ret = false;
+  }
+
+  if (ret) {
+    if (!HASHMAP_CONTRACT_STATE_DB) {
+      ContractStorage::GetContractStorage().GetStateDB().commit();
+
+      for (auto i : *m_addressToAccount) {
```

Comment:
I think this might need to be `auto&` to avoid copy-constructing another object in `i`